### PR TITLE
Draft: Develop

### PR DESCRIPTION
**Pull request by Pnom**

### CODE REVIEW

File: docs/obis-observer/commands/GetMeterArchiveState.md [hunk 1]
```diff
@@ -1,27 +1,33 @@
...
-| ---- | ------------------------------------ | ----------------------------------------------------------------------- |
-| `1`  | `byte`                               | command id = `0x28`                                                     |
-| `1`  | [Request ID](../types.md#request-id) | request/response unique identifier                                      |
-| `1`  | `byte`                               | archive type: <br> `1` - archive profile 1 <br> `2` - archive profile 2 |
+| Size | Type                                 | Field                                                   |
+| ---- | ------------------------------------ | ------------------------------------------------------- |
+| `1`  | `byte`                               | command id = `0x7c`                                     |
+| `1`  | `byte`                               | command size                                            |
+| `1`  | [Request ID](../types.md#request-id) | request/response unique identifier                      |
+| `1`  | [Meter id](../types.md#meter-id)     | meter unique identifier                                 |
```
Comment (DarkPark) [inline docs/obis-observer/commands/GetMeterArchiveState.md:15]:
```suggestion
| `1`  | [Meter ID](../types.md#meter-id)     | meter unique identifier                                 |
```

File: docs/obis-observer/commands/GetMeterArchiveState.md [hunk 2]
```diff
@@ -30,40 +36,43 @@ Message hex dump: `28 79 02`
...
 | newest record time | `2023.06.28 15:15:02 GMT` | `0x2c2f0af6` |
 
-Message hex dump: `29 02 00 00 00 51 2c 2d ea ae 2c 2f 0a f6`
+Message hex dump: `7b 0d 02 00 00 00 51 2c 2d ea ae 2c 2f 0a f6`
 
 
 ## See also
 
 * [Request ID](../types.md#request-id)
+* [Meter id](../types.md#meter-id)
```
Comment (DarkPark) [inline docs/obis-observer/commands/GetMeterArchiveState.md:77]:
```suggestion
* [Meter ID](../types.md#meter-id)
```

File: docs/obis-observer/types.md [hunk 3]
```diff
@@ -3,11 +3,32 @@
 Commands use many specific data types packed to bytes.
 This document describes these types and structuring principles.
 
+# Meter profile
 
-## Short name
+Meter profile contains all data which we need to observe, archive and report the OBIS content.
+- archive settings.
```
Comment (DarkPark) [inline docs/obis-observer/types.md:9]: either from upper letter or not a sentence

File: docs/obis-observer/types.md [hunk 4]
```diff
@@ -3,11 +3,32 @@
 Commands use many specific data types packed to bytes.
 This document describes these types and structuring principles.
 
+# Meter profile
```
Comment (DarkPark) [inline docs/obis-observer/types.md:6]: something is wrong with hierarchy here and below

File: docs/obis-observer/obis-id-list.md [hunk 5]
```diff
@@ -1,16 +1,16 @@
...
+`1`-`199` - [list of static id](#static) that can't be changed.
+`210`-`235` - [list of reassignable id](#reassignable).
 
 
 ## Static
 
 ### General
 
-| Short name | OBIS code | Description                                                             |
+| OBIS id  | OBIS code | Description                                                             |
```
Comment (DarkPark) [inline docs/obis-observer/obis-id-list.md:13]: need to fix formatting in this file

File: docs/obis-observer/message.md [hunk 6]
```diff
@@ -18,3 +18,22 @@ Devices send and receive messages in the following format:
...
+
+## Commands
+
+<table>
+    <tbody>
+        <tr align="center">
+            <td>Command id</td>
+        </tr>
+        <tr align="center">
+            <td>command data size</td>
```
Comment (DarkPark) [inline docs/obis-observer/message.md:30]:
```suggestion
            <td>Command data size</td>
```

?

and below

File: docs/obis-observer/commands/GetObisInfo.md [hunk 7]
```diff
@@ -1,30 +1,34 @@
...
-| `1`  | `byte`                               | command id = `0x0b`                |
-| `1`  | [Request ID](../types.md#request-id) | request/response unique identifier |
-| `1`  | [Short name](../types.md#short-name) | short name                         |
+| Size | Type                                             | Field                              |
+| ---- | ------------------------------------------------ | ---------------------------------- |
+| `1`  | `byte`                                           | command id = `0x4a`                |
+| `1`  | `byte`                                           | command size                       |
+| `1`  | [Request ID](../types.md#request-id)             | request/response unique identifier |
+| `1`  | [Meter profile id](../types.md#meter-profile-id) | meter profile unique identifier    |
+| `1`  | [OBIS id](../types.md#obis-id)                   | OBIS unique identifier             |
```
Comment (DarkPark) [inline docs/obis-observer/commands/GetObisInfo.md:16]:
```suggestion
| `1`  | [OBIS ID](../types.md#obis-id)                   | OBIS unique identifier             |
```

?

File: docs/obis-observer/commands/GetObserverCapabilities.md [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+| ------------------------- | ----- | ------ |
+| command id                | `26`  | `0x1a` |
+| command size              | `5`   | `0x05` |
+| request id                | `7`   | `0x07` |
+| max meter profiles number | `8`   | `0x08` |
+| max meter number          | `8`   | `0x08` |
+| max OBIS profiles number  | `8`   | `0x08` |
+| is multi-mode supported   | `1`   | `0x01` |
+
+Message hex dump: `1a 05 08 08 08 01`
```
Comment (DarkPark) [inline docs/obis-observer/commands/GetObserverCapabilities.md:55]: request id is missing

File: docs/obis-observer/commands/GetMeterArchiveState.md [hunk 9]
```diff
@@ -1,27 +1,33 @@
...
+| command id   | `124` | `0x7c` |
+| command size | `3`   | `0x03` |
+| request id   | `41`  | `0x29` |
+| meter id     | `3`   | `0x03` |
+| archive type | `1`   | `0x01` |
 
-Message hex dump: `28 79 02`
+
+
+Message hex dump: `7c 03 29 03 01`
```
Comment (DarkPark) [inline docs/obis-observer/commands/GetMeterArchiveState.md:30]: one empty line before is ok

File: docs/obis-observer/commands/RemoveMeter.md [hunk 10]
```diff
@@ -0,0 +1,79 @@
...
+#### success:
+
+| Field        | Value | Hex    |
+| ------------ | ----- | ------ |
+| command id   | `115` | `0x73` |
+| command size | `2`   | `0x02` |
+| request id   | `156` | `0x9c` |
+| result code  | `OK`  | `0x00` |
+
+Message hex dump: `71 02 9c 00`
```
Comment (DarkPark) [inline docs/obis-observer/commands/RemoveMeter.md:53]:
```suggestion
Message hex dump: `73 02 9c 00`
```

?

and below

File: docs/obis-observer/commands/RemoveMeterProfile.md [hunk 11]
```diff
@@ -0,0 +1,67 @@
...
+### Examples
+
+| Field            | Value | Hex    |
+| ---------------- | ----- | ------ |
+| command id       | `98`  | `0x62` |
+| command size     | `2`   | `0x02` |
+| request id       | `18`  | `0x12` |
+| meter profile id | `2`   | `0x02` |
+
+Message hex dump: `62 02 18 02`
```
Comment (DarkPark) [inline docs/obis-observer/commands/RemoveMeterProfile.md:26]:
```suggestion
Message hex dump: `62 02 12 02`
```

?

File: docs/obis-observer/commands/RemoveMeterProfile.md [hunk 12]
```diff
@@ -0,0 +1,67 @@
...
+#### success:
+
+| Field        | Value | Hex    |
+| ------------ | ----- | ------ |
+| command id   | `99`  | `0x63` |
+| command size | `2`   | `0x02` |
+| request id   | `7`   | `0x07` |
+| result code  | `OK`  | `0x00` |
+
+Message hex dump: `26 02 07 00`
```
Comment (DarkPark) [inline docs/obis-observer/commands/RemoveMeterProfile.md:52]:
```suggestion
Message hex dump: `63 02 07 00`
```

?

File: docs/obis-observer/commands/RemoveMeterProfile.md [hunk 13]
```diff
@@ -0,0 +1,67 @@
...
+## Request
+
+### Format
+
+| Size | Type                                             | Field                              |
+| ---- | ------------------------------------------------ | ---------------------------------- |
+| `1`  | `byte`                                           | command id = `0x62`                |
+| `1`  | `byte`                                           | command size                       |
+| `1`  | [Request ID](../types.md#request-id)             | request/response unique identifier |
+| `1`  | [Meter profile id](../types.md#meter-profile-id) | meter profile unique identifier    |
```
Comment (DarkPark) [inline docs/obis-observer/commands/RemoveMeterProfile.md:14]:
```suggestion
| `1`  | [Meter profile ID](../types.md#meter-profile-id) | meter profile unique identifier    |
```

?

File: docs/obis-observer/commands/SetSerialPort.md [hunk 14]
```diff
@@ -93,36 +96,46 @@ Message hex dump: `13 34 05 08 05`
...
-Message hex dump: `14 20 01`
+| Result code | Description                       |
+| ----------- | --------------------------------- |
+| `0`         | Ok. The Operation was successful. |
+| `2`         | Wrong arguments.                  |
 
 
 ## See also
 
 * [Request ID](../types.md#request-id)
```
Comment (DarkPark) [inline docs/obis-observer/commands/SetSerialPort.md:140]: some links are missing

File: docs/obis-observer/commands/AddMeterProfile.md [hunk 15]
```diff
@@ -0,0 +1,71 @@
...
+#### add meter profile with id `32`:
+
+| Field            | Value | Hex    |
+| ---------------- | ----- | ------ |
+| command id       | `96`  | `0x60` |
+| command size     | `2`   | `0x02` |
+| request id       | `3`   | `0x03` |
+| meter profile id | `32`  | `0x20` |
+
+Message hex dump: `40 02 03 20`
```
Comment (DarkPark) [inline docs/obis-observer/commands/AddMeterProfile.md:29]:
```suggestion
Message hex dump: `60 02 03 20`
```

?

File: docs/obis-observer/commands/GetMeterArchiveState.md (hunk 2 above)
Comment (Pnom) [reply docs/obis-observer/commands/GetMeterArchiveState.md]: did it

File: docs/obis-observer/commands/GetMeterArchiveState.md (hunk 9 above)
Comment (Pnom) [reply docs/obis-observer/commands/GetMeterArchiveState.md:28]: did it

File: docs/obis-observer/commands/GetMeterDate.md [hunk 16]
```diff
@@ -0,0 +1,58 @@
...
+## Request
+
+### Format
+
+| Size | Type                                 | Field                              |
+| ---- | ------------------------------------ | ---------------------------------- |
+| `1`  | `byte`                               | command id = `0x7a`                |
+| `1`  | `byte`                               | command size                       |
+| `1`  | [Request ID](../types.md#request-id) | request/response unique identifier |
+| `1`  | [Meter id](../types.md#meter-id)     | meter unique identifier            |
```
[the remark this answers is not in the document]
Comment (Pnom) [reply docs/obis-observer/commands/GetMeterDate.md]: did it

File: docs/obis-observer/types.md (hunk 3 above)
Comment (Pnom) [reply docs/obis-observer/types.md]: did it

File: docs/obis-observer/types.md (hunk 4 above)
Comment (Pnom) [reply docs/obis-observer/types.md]: did it

File: docs/obis-observer/commands/GetObisInfo.md (hunk 7 above)
Comment (Pnom) [reply docs/obis-observer/commands/GetObisInfo.md]: did it

File: docs/obis-observer/commands/GetObserverCapabilities.md (hunk 8 above)
Comment (Pnom) [reply docs/obis-observer/commands/GetObserverCapabilities.md]: did it

File: docs/obis-observer/commands/GetSingleMode.md [hunk 17]
```diff
@@ -0,0 +1,57 @@
...
+| request id   | `7`   | `0x07` |
+| single mode  | `1`   | `0x01` |
+
+Message hex dump: `0e 02 07 01`
+
+
+## See also
+
+* [Request ID](../types.md#request-id)
+* [Single mode](../single-mode.md)
```
[the remark this answers is not in the document]
Comment (Pnom) [reply docs/obis-observer/commands/GetSingleMode.md]: did it

File: docs/obis-observer/commands/RemoveMeter.md (hunk 10 above)
Comment (Pnom) [reply docs/obis-observer/commands/RemoveMeter.md]: did it

File: docs/obis-observer/commands/RemoveMeterProfile.md (hunk 11 above)
Comment (Pnom) [reply docs/obis-observer/commands/RemoveMeterProfile.md]: did it

File: docs/obis-observer/commands/RemoveMeterProfile.md (hunk 12 above)
Comment (Pnom) [reply docs/obis-observer/commands/RemoveMeterProfile.md]: did it

File: docs/obis-observer/commands/RemoveMeterProfile.md (hunk 13 above)
Comment (Pnom) [reply docs/obis-observer/commands/RemoveMeterProfile.md]: did it

File: docs/obis-observer/commands/SetMeterArchiveProfile.md [hunk 18]
```diff
@@ -0,0 +1,84 @@
...
+#### success:
+
+| Field        | Value | Hex    |
+| ------------ | ----- | ------ |
+| command id   | `105` | `0x69` |
+| command size | `2`   | `0x02` |
+| request id   | `156` | `0x9c` |
+| result code  | `0`   | `0x00` |
+
+Message hex dump: `10 9c 00`
```
[the remark this answers is not in the document]
Comment (Pnom) [reply docs/obis-observer/commands/SetMeterArchiveProfile.md]: did it

File: docs/obis-observer/commands/AddMeterProfile.md (hunk 15 above)
Comment (Pnom) [reply docs/obis-observer/commands/AddMeterProfile.md]: did it

File: docs/obis-observer/types.md [hunk 19]
```diff
@@ -3,11 +3,33 @@
...
+- custom OBIS table. Setup obis id for the custom OBIS code, see [reassignable table](./obis-id.md#reassignable)
+- table of the [OBIS profiles](#obis-profile). They describe how to read, archive and report the specific OBIS
+
+### Meter profile ID
+
+A `1`-byte unique number which used to represent the specific meter profile.
+`0xff` considers as invalid meter profile ID.
+
+## Meter
+s
```
Comment (DarkPark) [inline docs/obis-observer/types.md:19]: typo?

File: docs/obis-observer/types.md [hunk 20]
```diff
@@ -3,11 +3,33 @@
...
+- meter address. The string 32 characters maximum. Usually equals to the serial number of the meter.
+Only one meter with meter ID `0` have empty address and works only in [single mode](./singleMultiMode.md#single-mode).
+- [meter profile ID](#meter-profile-id). Link to meter profile to use.
+
+### Meter ID
+
+A `1`-byte number will be used in all messages to represent the specific meter.
+`0xff` considers as invalid meter ID.
+
+## Obis id
```
Comment (DarkPark) [inline docs/obis-observer/types.md:29]:
```suggestion
## Obis ID
```

? or

```suggestion
## OBIS ID
```

File: docs/obis-observer/types.md (hunk 19 above)
Comment (Pnom) [reply docs/obis-observer/types.md]: fixed

File: docs/obis-observer/types.md (hunk 3 above)
Comment (Pnom) [reply docs/obis-observer/types.md]: fixed

**Review by DarkPark — DISMISSED** [review]

**Review by DarkPark — DISMISSED** [review]

File: docs/obis-observer/commands/GetArchiveState.md [hunk 21]
```diff
@@ -7,25 +7,29 @@ Request/response to get the archive state for the specific meter, including the
...
-| Size | Type                                 | Field                                                   |
-| ---- | ------------------------------------ | ------------------------------------------------------- |
-| `1`  | `byte`                               | command id = `0x7c`                                     |
-| `1`  | `byte`                               | command size                                            |
-| `1`  | [Request ID](../types.md#request-id) | request/response unique identifier                      |
-| `1`  | [Meter ID](../types.md#meter-id)     | meter unique identifier                                 |
-| `1`  | `byte`                               | archive type: <br> `1` - archive 1 <br> `2` - archive 2 |
+| Size | Type                                 | Mandatory/optional | Field                                                   |
+| ---- | ------------------------------------ | ------------------ | ------------------------------------------------------- |
+| `1`  | `byte`                               | Mandatory          | command id = `0x0f`                                     |
```
Comment (DarkPark) [inline docs/obis-observer/commands/GetArchiveState.md:12]:
```suggestion
| `1`  | `byte`                               | mandatory          | command id = `0x0f`                                     |
```
?

as it's not a sentence

**Review by DarkPark — APPROVED** [review]

File: docs/obis-observer/commands/GetObisIdList.md [hunk 22]
```diff
@@ -7,18 +7,29 @@ Request/response to get the OBIS ID list for the specific OBIS code and meter pr
...
 
 ### Examples
 
-#### get OBIS ID for OBIS `0.9.1`:
+#### get OBIS ID list:
+
+| Field            | Value                  | Hex          |
+| ---------------- | ---------------------- | ------------ |
+| command id       | `64`                   | `0x40`       |
+| command size     | `2`                    | `0x06`       |
```
Comment (DarkPark) [inline docs/obis-observer/commands/GetObisIdList.md:26]:
```suggestion
| command size     | `2`                    | `0x02`       |
```

?

File: docs/obis-observer/commands/GetObisIdList.md [hunk 23]
```diff
@@ -7,18 +7,29 @@ Request/response to get the OBIS ID list for the specific OBIS code and meter pr
...
-| ----- | ------------------------------------------------ | ---------------------------------- |
-| `1`   | `byte`                                           | command id = `0x40`                |
-| `1`   | `byte`                                           | command size                       |
-| `1`   | [Request ID](../types.md#request-id)             | request/response unique identifier |
-| `1`   | [Meter profile ID](../types.md#meter-profile-id) | meter profile unique identifier    |
-| `3-7` | [OBIS](../types.md#obis)                         | OBIS code                          |
+| Size  | Type                                             | Mandatory/optional | Field                              |
+| ----- | ------------------------------------------------ | ------------------ | ---------------------------------- |
+| `1`   | `byte`                                           | mandatory          | command id = `0x40`                |
+| `1`   | `byte`                                           | mandatory          | command size                       |
```
Comment (DarkPark) [inline docs/obis-observer/commands/GetObisIdList.md:13]: command size 2+ ?

File: docs/obis-observer/commands/AddObisProfile.md [hunk 24]
```diff
@@ -61,11 +61,11 @@ Message hex dump: `45 02 07 00`
...
 
-| Result code | Description                                                |
-| ----------- | ---------------------------------------------------------- |
-| `0`         | Ok. The Operation was successful.                          |
-| `6`         | The OBIS profile id table full. Unable to add new entries. |
-| `10`        | The meter profile not found.                               |
+| Result code | Description                       |
+| ----------- | --------------------------------- |
+| `0`         | Ok. The Operation was successful. |
+| `5`         | Obis profile allocation failed.   |
```
Comment (DarkPark) [inline docs/obis-observer/commands/AddObisProfile.md:67]: OBIS ?

File: docs/obis-observer/commands/SetObisId.md [hunk 25]
```diff
@@ -59,12 +59,12 @@ Message hex dump: `43 02 14 00`
...
-| ----------- | -------------------------------------------------- |
-| `0`         | Ok. The Operation was successful.                  |
-| `3`         | Forbidden to reassign the static OBIS ID.          |
-| `4`         | The OBIS ID table full. Unable to add new entries. |
-| `10`        | The meter profile not found.                       |
+| Result code | Description                               |
+| ----------- | ----------------------------------------- |
+| `0`         | Ok. The Operation was successful.         |
+| `3`         | Forbidden to reassign the static OBIS ID. |
+| `4`         | Obis id allocation failed.                |
```
Comment (DarkPark) [inline docs/obis-observer/commands/SetObisId.md:66]: OBIS ?

Comment (Pnom) [reply docs/obis-observer/commands/SetObisId.md:66]: fixed

File: docs/obis-observer/commands/AddObisProfile.md (hunk 24 above)
Comment (Pnom) [reply docs/obis-observer/commands/AddObisProfile.md]: fixed

File: docs/obis-observer/commands/GetObisIdList.md (hunk 23 above)
Comment (Pnom) [reply docs/obis-observer/commands/GetObisIdList.md]: fixed

File: docs/obis-observer/commands/GetMeterReadoutState.md [hunk 26]
```diff
@@ -9,17 +9,24 @@ Request/response to get the readout related state and statistic.
...
 | `1`  | [Request ID](../types.md#request-id) | request/response unique identifier |
+| `1`  | [Meter ID](../types.md#meter-id)     | meter unique identifier            |
 
 ### Examples
 
-| Field      | Value | Hex    |
-| ---------- | ----- | ------ |
-| command id | `38`  | `0x26` |
-| request id | `18`  | `0x12` |
+#### get readout state for meter with id `8`
```
Comment (DarkPark) [inline docs/obis-observer/commands/GetMeterReadoutState.md:19]:
```suggestion
#### get readout state for meter with id `8`:
```

File: docs/obis-observer/commands/ReadArchive.md [hunk 27]
```diff
@@ -7,59 +7,107 @@ Request/response to get archive data.
...
+If meter id field is not provided, the observer will respond with records from all meters.
+
+The date field specifies a specific time within a day. For example, if the field indicates `14:10`, the response will include all the archived data captured from the start of the nearest archive period to `14:10` until the end of that same archive period.
+If date field is provided, the observer will respond with records with that specified date.
+If date field is not provided, the observer will respond with records regardless of the date.
+
 
 ### Examples
 
+#### request without the meter id and date
```
Comment (DarkPark) [inline docs/obis-observer/commands/ReadArchive.md:30]:
```suggestion
#### request without the meter id and date:
```

File: docs/obis-observer/commands/ReadArchive.md [hunk 28]
```diff
@@ -7,59 +7,107 @@ Request/response to get archive data.
...
+| ------------ | ----------- | ------ |
+| command id   | `17`        | `0x11` |
+| command size | `3`         | `0x03` |
+| request id   | `33`        | `0x21` |
+| archive type | `Archive 1` | `0x01` |
+| start index  | `0`         | `0x00` |
+
+Message hex dump: `11 03 21 01 00`
+
+#### request with the meter id
```
Comment (DarkPark) [inline docs/obis-observer/commands/ReadArchive.md:42]:
```suggestion
#### request with the meter id:
```

File: docs/obis-observer/commands/ReadArchive.md [hunk 29]
```diff
@@ -7,59 +7,107 @@ Request/response to get archive data.
...
+| command id   | `17`        | `0x11` |
+| command size | `4`         | `0x04` |
+| request id   | `33`        | `0x21` |
+| archive type | `Archive 1` | `0x01` |
+| start index  | `0`         | `0x00` |
+| meter id     | `2`         | `0x02` |
+
+Message hex dump: `11 04 21 01 00 02`
+
+#### request with the meter id and date
```
Comment (DarkPark) [inline docs/obis-observer/commands/ReadArchive.md:55]:
```suggestion
#### request with the meter id and date:
```

File: docs/obis-observer/commands/SetObisId.md (hunk 25 above)
Comment (DarkPark) [reply docs/obis-observer/commands/SetObisId.md:66]: see no changes

**Review by DarkPark — CHANGES_REQUESTED** [review]

File: docs/obis-observer/commands/SetObisId.md (hunk 25 above)
Comment (Pnom) [reply docs/obis-observer/commands/SetObisId.md]: sorry. fixed now

File: docs/obis-observer/commands/GetMeterReadoutState.md (hunk 26 above)
Comment (Pnom) [reply docs/obis-observer/commands/GetMeterReadoutState.md]: fixed

File: docs/obis-observer/commands/ReadArchive.md (hunk 29 above)
Comment (Pnom) [reply docs/obis-observer/commands/ReadArchive.md]: fixed

File: docs/obis-observer/commands/ReadArchive.md (hunk 27 above)
Comment (Pnom) [reply docs/obis-observer/commands/ReadArchive.md]: there is no such text more

File: docs/obis-observer/commands/ReadArchive.md (hunk 28 above)
Comment (Pnom) [reply docs/obis-observer/commands/ReadArchive.md]: There is no such text now.

File: docs/obis-observer/commands/GetMeterArchiveProfile.md [hunk 30]
```diff
@@ -0,0 +1,75 @@
...
+| `3`         | Format error.                |
+| `10`        | The meter profile not found. |
+
+## See also
+
+* [Request ID](../types.md#request-id)
+* [Meter profile ID](../types.md#meter-profile-id)
+* [Result code](../types.md#result-code)
+* [Error](./uplink/Error.md)
+*
```
Comment (DarkPark) [inline docs/obis-observer/commands/GetMeterArchiveProfile.md:75]:
```suggestion
```

Comment (Pnom) [reply docs/obis-observer/commands/GetMeterArchiveProfile.md]: fixed

File: docs/obis-observer/commands/GetObisInfoList.md [hunk 31]
```diff
@@ -0,0 +1,79 @@
...
+| `1`  | `byte`                               | is list completed                  |
+| `1`  | [OBIS ID](../types.md#obis-id)       | OBIS ID `1`                        |
+| `1`  | [OBIS info flag](#obis-info-flag)    | OBIS info flag for the OBIS ID `1` |
+| ...  | ...                                  | ...                                |
+| `1`  | [OBIS ID](../types.md#obis-id)       | OBIS ID `N`                        |
+| `1`  | [OBIS info flag](#obis-info-flag)    | OBIS info flag for the OBIS ID `N` |
+
+#### OBIS info flag
+
+| Bit 7 - Bit 2 | Bit 1                                     | Bit 0                                       |
```
[the remark this answers is not in the document]
Comment (Pnom) [reply docs/obis-observer/commands/GetObisInfoList.md]: fixed

**Pull request merged**